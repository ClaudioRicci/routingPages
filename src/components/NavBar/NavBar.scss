@import "./../../variables.scss";

ul {
  display: flex;
  flex-direction: row;

  li {
    list-style: none;
    padding: 0.5rem;
    width: 10rem;

    width: 70px;
    height: 0;
    border-right: 20px solid transparent;
    border-bottom: 50px solid $skyblue;

    a {
      color: #000;
      font-weight: bold;
      text-decoration: none;
    }
  }
}
