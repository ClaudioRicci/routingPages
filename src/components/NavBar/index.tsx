import React from "react";
import { BrowserRouter as Router, Route, Switch, Link } from "react-router-dom";
import uuid from "uuid";
import dataAPI from "../../data/api";
import Panel from "../../components/Panel";
import "./NavBar.scss";

const NavBar = () => (
  <Router>
    <div className="App">
      <ul>
        {dataAPI.map(item => {
          return (
            <li key={uuid.v4()}>
              <Link
                to={{
                  pathname: item.path,
                  state: {
                    title: item.pageTitle,
                    description: item.pageDescription,
                    tableColumms: item.table.columns,
                    dataUrl: item.table.apiEndpoint
                  }
                }}
              >
                {item.linkTitle}
              </Link>
            </li>
          );
        })}
      </ul>
      <Switch>
        <Route exact path="/:page" component={Panel} />
      </Switch>
    </div>
  </Router>
);

export default NavBar;
