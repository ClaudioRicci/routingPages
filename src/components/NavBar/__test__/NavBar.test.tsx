import React from "react";
import ReactDOM from "react-dom";
import NavBar from "..";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("Renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<NavBar></NavBar>, div);
});

test("Renders NavBar Correctly", done => {
  const { getByTestId } = render(<NavBar label="Close" />);
  expect(getByTestId("NavBar")).toHaveTextContent("Page A");
  done();
});

it("Matches snapshot", () => {
  const tree = renderer.create(<NavBar></NavBar>).toJSON();
  expect(tree).toMatchSnapshot();
});
