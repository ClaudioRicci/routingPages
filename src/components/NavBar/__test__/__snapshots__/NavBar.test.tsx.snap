// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
Array [
  <ul
    aria-label="Main Navigation"
    data-testid="NavBar"
    role="navigation"
  >
    <li>
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="/page-a"
        onClick={[Function]}
      >
        Page A
      </a>
    </li>
    <li>
      <a
        href="/page-b"
        onClick={[Function]}
      >
        Page B
      </a>
    </li>
    <li>
      <a
        href="/page-c"
        onClick={[Function]}
      >
        Page C
      </a>
    </li>
  </ul>,
  <main
    data-testid="Home"
    role="main"
  >
    <h1>
      Home Page
    </h1>
    <p>
      Please select a page from the tabs above
    </p>
  </main>,
]
`;
